devito/finite_differences/derivative.py

```diff
@@ -92,19 +93,95 @@ class Derivative(sympy.Derivative, Differentiable, Pickable):
                    'x0', 'method', 'weights')
 
     def __new__(cls, expr, *dims, **kwargs):
+        # TODO: Delete this
+        if kwargs.get('preprocessed', False):
+            from warnings import warn
+            warn('I removed the `preprocessed` kwarg')
+
         if type(expr) is sympy.Derivative:
             raise ValueError("Cannot nest sympy.Derivative with devito.Derivative")
         if not isinstance(expr, Differentiable):
-            raise ValueError("`expr` must be a Differentiable object")
+            try:
+                expr = diffify(expr)
+            except Exception as e:
+                raise ValueError("`expr` must be a Differentiable object") from e
+
+        # Validate `dims`. It can be:
+        # - a single Dimension ie: x
+        # - an iterable of Dimensions ie: (x, y)
+        # - a single tuple of Dimension and order ie: (x, 2)
+        # - or an iterable of Dimension, order ie: ((x, 2), (y, 2))
+        # - any combination of the above ie: ((x, 2), y, x, (z, 3))
+        if len(dims) == 0:
+            raise ValueError('Expected Dimension w.r.t. which to differentiate')
+        elif len(dims) == 1 and isinstance(dims[0], Iterable) and len(dims[0]) != 2:
+            # Iterable of Dimensions
+            raise ValueError(f'Expected `(dim, deriv_order)`, got {dims[0]}')
+        elif len(dims) == 2 and not isinstance(dims[1], Iterable) and is_integer(dims[1]):
+            # special case of single dimension and order
+            dims = (dims, )
```

Comment:
Just use `as_tuple`

Reply:
That doesn't work in this case, `dim` is already a `tuple`, but I want a nested tuple